### PR TITLE
Added example of how to run Hugging Face models through Ollama using TC

### PR DESCRIPTION
**Pull request by ilopezluna**
Added example of how to use `OllamaContainer` to run Hugging Face models

Co-authored-by: @kiview

### CODE REVIEW

File: examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceContainer.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                                .run("apt-get update && apt-get upgrade -y && apt-get install -y python3-pip")
+                                .run("pip install huggingface-hub")
+                                .build();
+                        })
+                );
+        }
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
```
Comment (kiview) [inline examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceContainer.java:42]:
```suggestion
    protected void containerIsStarted(InspectContainerResponse containerInfo, boolean reused) {
```

Support re-use.

File: examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceContainer.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        super.containerIsStarted(containerInfo);
```
Comment (kiview) [inline examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceContainer.java:43]:
```suggestion
        super.containerIsStarted(containerInfo, reused);
         if (reused) {
           return;
         }
```

Don't download the model on re-use.

**Review by kiview — CHANGES_REQUESTED** [review]
For adding reuse support, see suggestions (try if they compile like this 😅 ).

File: examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceContainer.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    new ImageFromDockerfile()
+                        .withDockerfileFromBuilder(builder -> {
+                            builder
+                                .from(this.getDockerImageName())
+                                .run("apt-get update && apt-get upgrade -y && apt-get install -y python3-pip")
+                                .run("pip install huggingface-hub")
+                                .build();
+                        })
+                );
+        }
```
Comment (eddumelendez) [inline examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceContainer.java:38]: I think we can move those commands to `containerIsStarted` and use `execInContainer` directly.

File: examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                    OllamaHuggingFaceContainer huggingFaceContainer = new OllamaHuggingFaceContainer(
+                        imageName,
+                        new OllamaHuggingFaceContainer.HuggingFaceModel(repository, model)
+                    )
+                ) {
+                    huggingFaceContainer.start();
+                    huggingFaceContainer.stop();
+                }
+                ollama.start();
+            }
```
Comment (eddumelendez) [inline examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceTest.java:40]: let's use the docker client API to check if the image exists, the code will be more readable.

**Review by eddumelendez — CHANGES_REQUESTED** [review]
Thanks @ilopezluna! Left some comments to improve the example.

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks @ilopezluna !

**Pull request merged**